src/Persistence/Wolverine.Marten/IMartenOp.cs

```diff
@@ -285,12 +317,24 @@ public override void Execute(IDocumentSession session)
 }
 
 public abstract class DocumentOp : IMartenOp
+{
+    public object Document { get; }
+
+    protected DocumentOp(object document)
+    {
+        Document = document;
+    }
+
+    public abstract void Execute(IDocumentSession session);
+}
+
+public abstract class DocumentsOp : IMartenOp
 {
     public object[] Documents { get; }
 
-    protected DocumentOp(params object[] documents)
+    protected DocumentsOp(IEnumerable<object> documents)
```

Comment:
I suggest keeping the `params` keyword to make the usage nicer.

```suggestion
    protected DocumentsOp(params object[] documents)
```